docs/proposal/2022-11-22-botkube-plugins-integration.md

```diff
@@ -0,0 +1,542 @@
+
+<details>
+  <summary>Discarded alternative</summary>
+
+### The `plugins` object syntax
+
+Configuration defined in `values.yaml`
+
+```yaml
+plugins:
+  repositories:
+    botkube: "https://github.com/kubeshop/botkube/releases/download/0.17.0/plugins-index.yaml"
+    mszostok: "https://github.com/mszostok/botkube-plugins/releases/download/latest/index.yaml"
+  # alternative, which is more extensible but doesn't allow to enforce unique name OOTB
+  repositories:
+    - name: botkube
+      url: "https://github.com/kubeshop/botkube/releases/download/0.17.0/plugins-index.yaml"
+    - name: mszostok
+      url: "https://github.com/mszostok/botkube-plugins/releases/download/latest/index.yaml"
+  # alternative, which is more extensible and allow to enforce unique name OOTB
+  repositories:
+    botkube:
+      url: "https://github.com/kubeshop/botkube/releases/download/0.17.0/plugins-index.yaml"
+    mszostok:
+      url: "https://github.com/mszostok/botkube-plugins/releases/download/latest/index.yaml"
+
+  enabled: # or 'install:'
+    - name: botkube/kubernetes
+      version: v0.17.0
+    - name: mszostok/kubectl
+      # version not provided, so use latest
+```
+
+We could also introduce the CRD, for example:
+
+```yaml
+apiVersion: plugins.botkube.io/v1alpha1
+kind: ClusterPluginConfiguration
+metadata:
+  name: bk-plugins
+spec:
+  repositories:
+    official: "https://github.com/kubeshop/botkube/releases/download/0.17.0/plugins-index.yaml"
+  enabled:
+    - name: official/kubernetes
+      version: v0.17.0
+```
+
+### How a given plugins is configured
+
+- Create dedicated instance (subprocess) for each different plugin config. This won't scale well.
+- Passing configuration data when starting a given plugin:
+  - Pass as serialized JSON/YAML using cmd flags.
+  - Save to file and specify flag with path location.
+- Add additional method like `Initialize()/SetConfig()`
+
+### Defining executors aliases
+
+One option is to define them in the index file for a given executor or source:
+
+```yaml
+entries:
+  - name: "kubectl"
+    type: "executor"
+    description: "Kubectl executor plugin."
+    version: "v1.0.0"
+    aliases: [ "kc", "k" ]
+    urls:
+      - url: https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-darwin-amd64
+        platform:
+          architecture: amd64
+          os: darwin
+      - url: https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-darwin-arm64
+        platform:
+          architecture: arm64
+          os: darwin
+      - url: https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-linux-amd64
+        platform:
+          architecture: amd64
+          os: linux
+      - url: https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-linux-arm64
+        platform:
+          architecture: arm64
+          os: linux
+```
+
+Optionally
+</details>
```

Comment:
I think it will be good to mention also the alternative idea with defining all the command prefixes by the user in a single field (`aliases/prefixes: [kubectl, k, kc]`) we were discussing recently 👍

Reply:
added